lib/pf/ConfigStore/Switch.pm

```diff
@@ -18,108 +18,34 @@ use pf::log;
-        }
-    ],
-    -oncachereload => [
-        on_cached_overlay_reload => sub {
-            my ( $config, $name ) = @_;
-            my $data = $config->fromCacheForDataUntainted("SwitchConfig");
-            if($data) {
-                %SwitchConfig = %$data;
-            } else {
-                #if not found then repopulate switch
-                $config->_callFileReloadCallbacks();
-            }
-        },
-    ]
-);
-
-sub populateSwitchConfig {
-    my ( $config, $name ) = @_;
-    $config->toHash( \%SwitchConfig );
-    $config->cleanupWhitespace( \%SwitchConfig );
-    foreach my $switch ( values %SwitchConfig ) {
-
-        # transforming uplink and inlineTrigger to arrays
-        foreach my $key (qw(uplink inlineTrigger)) {
-            my $value = $switch->{$key} || "";
-            $switch->{$key} = [ split /\s*,\s*/, $value ];
-        }
+=cut
 
-        # transforming vlans and roles to hashes
-        my %merged = ( Vlan => {}, Role => {}, AccessList => {} );
-        foreach my $key ( grep {/(Vlan|Role|AccessList)$/} keys %{$switch} ) {
-            next unless my $value = $switch->{$key};
-            if ( my ( $type_key, $type ) = ( $key =~ /^(.+)(Vlan|Role|AccessList)$/ ) ) {
-                $merged{$type}{$type_key} = $value;
-            }
-        }
-        $switch->{roles}        = $merged{Role};
-        $switch->{vlans}        = $merged{Vlan};
-        $switch->{access_lists} = $merged{AccessList};
-        $switch->{VoIPEnabled} = (
-            $switch->{VoIPEnabled} =~ /^\s*(y|yes|true|enabled|1)\s*$/i
-            ? 1
-            : 0
-        );
-        $switch->{mode} = lc( $switch->{mode} );
-        $switch->{'wsUser'} ||= $switch->{'htaccessUser'};
-        $switch->{'wsPwd'} ||= $switch->{'htaccessPwd'} || '';
-        foreach my $cli_default (qw(EnablePwd Pwd User)) {
-            $switch->{"cli${cli_default}"}
-              ||= $switch->{"telnet${cli_default}"};
-        }
-        foreach my $snmpDefault (
-            qw(communityRead communityTrap communityWrite version)) {
-            my $snmpkey = "SNMP" . ucfirst($snmpDefault);
-            $switch->{$snmpkey} ||= $switch->{$snmpDefault};
-        }
-    }
-    $SwitchConfig{'127.0.0.1'} = {
-        %{ $SwitchConfig{default} },
-        type              => 'PacketFence',
-        mode              => 'production',
-        uplink            => ['dynamic'],
-        SNMPVersionTrap   => '1',
-        SNMPCommunityTrap => 'public'
-    };
-    $config->cacheForData->set( "SwitchConfig", \%SwitchConfig );
-}
+=item commit
 
-sub _buildCachedConfig { $switches_cached_config }
+Repopulate the radius_nas table after commiting
 
-=head2 Methods
+=cut
 
-=over
+sub commit {
+    my ($self) = @_;
+    my ($result, $error) = $self->SUPER::commit();
+    pf::log::get_logger->info("commiting via Switch configstore");
+    freeradius_populate_nas_config( \%pf::SwitchFactory::SwitchConfig );
```

Comment:
I am asking just to make sure.
Would freeradius_populate_nas_config always get the latest version of pf::SwitchFactory::SwitchConfig?
Or is it possible it will get the old pf::SwitchFactory::SwitchConfig?